cnf-certification-test/platform/suite.go

```diff
@@ -176,8 +183,16 @@ func testTainted(env *provider.TestEnvironment) {
 	// We are expecting tainted nodes to be Nil, but only if:
 	// 1) The reason for the tainted node is contains(`module was loaded`)
 	// 2) The modules loaded are all whitelisted.
-	gomega.Expect(taintedNodes).To(gomega.BeNil())
-	gomega.Expect(errNodes).To(gomega.BeNil())
+	tnf.GomegaExpectSliceBeNil(taintedNodes)
+	tnf.GomegaExpectSliceBeNil(errNodes)
+
+	if tnf.IsUnitTest() {
```

Comment:
Honestly, I don't like these `if tnf.IsUnitTest() {` blocks for two reasons:
1. It's adding unit-testing related code into the function's normal/production code. I believe this "IsUnitTest()" or whatever variables for controlling the function's execution flow just for the sake of unit-testing should never exist inside the production code. That's precisely one of the advantages of mocking functions: this kind of tricks are not needed.
2. As you're mocking every external function here, it seems more natural to mock the tnf.GinkoFail or the tnf.GomegaExpect* functions to keep track of the failing scenarios of this tested function, as the normal execution flow for a failing tc should "die" on line 186 or 187.